### PR TITLE
fix: Aggregation without aggregation expressions should use correct result expressions

### LINKED ISSUE #176: Incorrect output when aggregation without aggregation functions
### Describe the bug

When aggregation operator (`HashAggregateExec`) has empty aggregation expressions, we only assign grouping by expressions but leave result aggregation empty. It is not correct. The result expressions are not always same as grouping by expressions.

For example, you can do `SELECT b, a, a + 1, b + 1 FROM table GROUP BY a, b`.

### Steps to reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

Closes #176.

### CODE REVIEW

File: spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala [hunk 1]
```diff
@@ -1783,6 +1783,13 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde {
         if (aggregateExpressions.isEmpty) {
           val hashAggBuilder = OperatorOuterClass.HashAggregate.newBuilder()
           hashAggBuilder.addAllGroupingExprs(groupingExprs.map(_.get).asJava)
+          val attributes = groupingExpressions.map(_.toAttribute) ++ aggregateAttributes
+          val resultExprs = resultExpressions.map(exprToProto(_, attributes))
+          if (resultExprs.exists(_.isEmpty)) {
+            emitWarning(s"Unsupported result expressions found in: ${resultExpressions}")
+            return None
+          }
```
Comment (viirya) [inline spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala:1791]: Even `aggregateExpressions` is empty, we still need to assign result expressions. This is the latest bug I found during fixing TPCDS query failures in SortMergeJoin work.

**Comment by viirya** [comment]
cc @sunchao @huaxingao

**Review by sunchao — APPROVED** [review]
LGTM (pending CI)

**Comment by viirya** [comment]
Merged. Thanks.

**Pull request merged**